### ISSUE TITLE
cannot allocate vector of size 69.8 Gb

### ISSUE THREAD

**Issue by metalp-epfl** [FarrellDay/miceRanger#6]
Hi, probably not really an issue due to the size of my dataset (110 mb of txt data, 1'000'000 observations x 14 variables) and the RAM/CPU of my computer (16 gb, 4 cores), but I report it as suggested by the package.

If I run such amount of data with 

  miceObj <- miceRanger(
    data
    , m=1
    , returnModels = FALSE
    , verbose=TRUE
  )

I get this error:

>Error in miceRanger(valsorey, m = 1, returnModels = FALSE, verbose = TRUE,  : 
  Evaluation failed with error <Error: cannot allocate vector of size 69.8 Gb
>. This is probably our fault - please open an issue at https://github.com/FarrellDay/miceRanger/issues with a reproduceable example.

I tried to reduce the number of trees to 50 and use "sample.fraction = 0.05" but I keep getting the same error. I did not try to run the code on a different machine (I'll soon have a 12 cores 64 gb ram at disposal) but for comparison I was able to get an imputation of my dataset using the package "missRanger" that does a similar job than miceRanger.

Is it a RAM problem of my computer or a limitation of the package? I can upload my data somewhere if needed.

**Comment by samFarrellDay** [comment]
That error is worded that way because, theoretically, all of the data and parameters should be formatted correctly before being sent to the imputation process. I assumed that any errors encountered there meant something wasn't caught early, but this memory issue would be an exception.

For now, see if using the ranger parameter `save.memory = TRUE` helps, I'll take a look today and see if there is any unnecessary memory usage.

**Comment by tsatsatsa** [comment]
Thank you for looking into it. Glad if my issues can help, this package is extremely promising for many environmental research fields! 
I'll do a test with sa.memory=TRUE and report it back. I'll have access to a 12 core 64gb RAM in the next days and I'll check on a different machine. In case you want to test it, the new dataset is similar to the one I shared with you last week (same 14 variables), but much longer. For testing purposes, you can basically copy/paste the observation x12 times.

**Comment by tsatsatsa** [comment]
I get the same error message with a 12 core CPU, 64 gb ram. The RAM was used up to 45GB before getting the error, thus not a memory limitation.

**Comment by Guiquanwang** [comment]
Truely, I got the same erro message

**Comment by samFarrellDay** [comment]
Yeah unfortunately I can't find a way to reduce the ranger model size. It is massive. See the code below:

```
library(ranger)

x <- rnorm(100000)
w <- rnorm(100000)
y <- x + w + rnorm(100000)
dat <- data.frame(x=x,w=w,y=y)

model <- ranger(y ~ x + w,data=dat)
model_sm <- ranger(y ~ x + w,data=dat,save.memory = TRUE)

format(object.size(model),units="MB")
format(object.size(model_sm),units="MB")
```

That model, which only has 100k rows and 2 independent variables, takes up an entire gigabyte of space. Apparently, setting `save.memory=TRUE` doesn't help. I'm going to talk to the ranger devs and see if they have any thoughts. The model is already deleted at each iteration [here](https://github.com/FarrellDay/miceRanger/blob/master/R/runIterations.R#L111), so the only thing I can think of is there is a memory leak in ranger or the predictions collected in each iteration truly add up to a large amount of memory, which I can't imagine being the case. I'll do a little more testing.

**Comment by samFarrellDay** [comment]
This _should_ be fixed by the latest update 1.4.0 which set the default max.depth parameter to 10. This greatly cut down on the memory usage without really affecting accuracy, at least in my testing. If anyone else is getting unfair memory usage, feel free to re-open this issue.
